test/test_fx.py

```diff
@@ -2258,6 +2259,251 @@ def test_get_torch_func_signature_exhaustive(self, device, dtype, op):
+        "celu": CONTROL_FLOW,
+        "cosine_embedding_loss": CONTROL_FLOW,
+        "cross_entropy": CONTROL_FLOW,
+        "ctc_loss": CONTROL_FLOW,
+        "dropout": CONTROL_FLOW,
+        "dropout2d": CONTROL_FLOW,
+        "dropout3d": CONTROL_FLOW,
+        "elu": CONTROL_FLOW,
+        "embedding": CONTROL_FLOW,
+        "embedding_bag": CONTROL_FLOW,
+        "feature_alpha_dropout": CONTROL_FLOW,
+        "fold": CONTROL_FLOW,
+        "gaussian_nll_loss": CONTROL_FLOW,
+        "glu": CONTROL_FLOW,
+        "grid_sample": CONTROL_FLOW,
+        "gumbel_softmax": CONTROL_FLOW,
+        "hardsigmoid": CONTROL_FLOW,
+        "hardswish": CONTROL_FLOW,
+        "hardtanh": CONTROL_FLOW,
+        'hinge_embedding_loss': CONTROL_FLOW,
+        'huber_loss': CONTROL_FLOW,
+        'interpolate': CONTROL_FLOW,
+        'kl_div': CONTROL_FLOW,
+        'l1_loss': CONTROL_FLOW,
+        'leaky_relu': CONTROL_FLOW,
+        'local_response_norm': CONTROL_FLOW,
+        'margin_ranking_loss': CONTROL_FLOW,
+        'mse_loss': CONTROL_FLOW,
+        'multi_head_attention_forward': CONTROL_FLOW,
+        'multi_margin_loss': CONTROL_FLOW,
+        'multilabel_margin_loss': CONTROL_FLOW,
+        'multilabel_soft_margin_loss': CONTROL_FLOW,
+        'nll_loss': CONTROL_FLOW,
+        'poisson_nll_loss': CONTROL_FLOW,
+        'relu': CONTROL_FLOW,
+        'relu6': CONTROL_FLOW,
+        'rrelu': CONTROL_FLOW,
+        'selu': CONTROL_FLOW,
+        'silu': CONTROL_FLOW,
+        'smooth_l1_loss': CONTROL_FLOW,
+        'soft_margin_loss': CONTROL_FLOW,
+        'threshold': CONTROL_FLOW,
+        'triplet_margin_loss': CONTROL_FLOW,
+        'triplet_margin_with_distance_loss': CONTROL_FLOW,
+        'unfold': CONTROL_FLOW,
+        'upsample': CONTROL_FLOW,
+
+        "upsample_bilinear": INTERPOLATE_ARGS_CONFLICT,
+        "upsample_nearest": INTERPOLATE_ARGS_CONFLICT,
+    }
+
+    # List of nn.functionals with Tensor inputs but not with type annotation
+    FUNCTIONALS_WITHOUT_ANNOTATION = (
+        "adaptive_max_pool1d",
+        "adaptive_max_pool2d",
+        "adaptive_max_pool3d",
+        "fractional_max_pool2d",
+        "fractional_max_pool3d",
+        "max_pool1d",
+        "max_pool2d",
+        "max_pool3d",
+        "gaussian_nll_loss",
+        "upsample",
+        "upsample_bilinear",
+        "upsample_nearest",
+    )
+
+    # For the operator like `in`, internal exception during the iteration
+    # can be raised on Python 3.8, rather than the error of `object is not iterable`
+    UNTRACEABLE_FUNCTIONALS_PY38 = {
+        "adaptive_max_pool1d": PROXY_ITERATED,
+        "adaptive_max_pool2d": PROXY_ITERATED,
+        "adaptive_max_pool3d": PROXY_ITERATED,
+        "fractional_max_pool2d": PROXY_ITERATED,
+        "fractional_max_pool3d": PROXY_ITERATED,
+        "max_pool1d": PROXY_ITERATED,
+        "max_pool2d": PROXY_ITERATED,
+        "max_pool3d": PROXY_ITERATED,
+
+        "group_norm": LEN_ERROR
+    }
```

Comment:
Has to implement this hack

Reply:
I don't 100% get the context here, can you paste a stack trace or something?

It's great that we're uncovering issues like this BTW, thanks!

Reply:
Use `adaptive_max_pool1d ` as an example. When we do `symbolic_trace` over it, the Error is different with different Python version.
- Python 3.8
```py
```
- Other (Use Python 3.6 as an example)
```py
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/erjia/workspace/pytorch-dev/py36/torch/fx/symbolic_trace.py", line 674, in symbolic_trace
    graph = tracer.trace(root, concrete_args)
  File "/Users/erjia/workspace/pytorch-dev/py36/torch/fx/symbolic_trace.py", line 419, in trace
    self.create_node('output', 'output', (self.create_arg(fn(*args)),), {},
  File "/Users/erjia/workspace/pytorch-dev/py36/torch/_jit_internal.py", line 395, in fn
    if arg_name in kwargs:
TypeError: argument of type 'Proxy' is not iterable
```

The error is caused here when `kwargs` is a Proxy object
https://github.com/pytorch/pytorch/blob/75eb026e07062e8539c3b6e96d6100ac9378a292/torch/_jit_internal.py#L395
When Proxy object is operated by `in`, `__iter__` function will be called and it will eventually raise a TraceError from the `iter` function of `TraceBase`. But, it's able to re-raise the internal TraceError with Python 3.8. It will raise `<object> is not iterable` with other python versions.

You can use the example in the note of main comment to verify this behavior.

And, I will update the comment here to make it easier to understand.